node/src/components/linear_chain_sync/event.rs

```diff
@@ -1,9 +1,16 @@
 use crate::types::{ActivationPoint, Block, BlockHash};
-use std::fmt::{Debug, Display};
+
+use casper_types::{PublicKey, U512};
+
+use std::{
+    collections::BTreeMap,
+    fmt::{Debug, Display},
+};
 
 #[derive(Debug)]
 pub enum Event<I> {
     Start(I),
+    GotInitialValidators(I, BTreeMap<PublicKey, U512>),
```

Comment:
Why do we need the `I` (node) here? Can you add a comment?

Reply:
Just to pass the parameter we got in `Start` - this event is basically an additional step between `Start` and what `Start` was doing, so the parameter from `Start` will still be useful.

Reply:
Why not add it to `self.peers_state` and remove it from this event?

Reply:
Mostly because I wanted to keep it as close as possible to the original code. I can do that, though.